### PR TITLE
Initial test for #7

### PR DESCRIPTION
**Pull request by AydinHassan**
Ping @barryvdh, @bencorlett

I tried to write a test to outline the issue in #7, however I cannot reproduce it.

Any chance you guys can hint at what you are doing differently?

I created 3 empty packages, `Package A` requires `Package B` & `Package C`.
The project requires all 3 packages explicitly.

### CODE REVIEW

**Comment by bencorlett** [comment]
I'll try reproduce it tomorrow. I can confirm it only effects private packages, be it coincidence or related I'm not sure. 

Sent from my iPhone

Please excuse my brevity

**Comment by barryvdh** [comment]
FYI, I also used private packages and those do seem to trigger the error.

**Comment by AydinHassan** [comment]
Hmm, that's strange. There is literally no different code paths for public and private packages.

File: tests/MagentoHackathon/Composer/FullStack/Regression/Issue007Test.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ *
+ *
+ *
+ *
+ */
```
Comment (nevvermind) [inline tests/MagentoHackathon/Composer/FullStack/Regression/Issue007Test.php]: Why this?

Comment (AydinHassan) [inline tests/MagentoHackathon/Composer/FullStack/Regression/Issue007Test.php]: c+p of the other test. Seems @Flyingmana likes to add this in.

Comment (Flyingmana) [inline tests/MagentoHackathon/Composer/FullStack/Regression/Issue007Test.php]: basic comment block added by phpstorm on file create

Comment (AydinHassan) [inline tests/MagentoHackathon/Composer/FullStack/Regression/Issue007Test.php]: OK, I've removed it

**Comment by Flyingmana** [comment]
still needed after the other PR got merged?

**Comment by Flyingmana** [comment]
sorry, should have looked more into it before commenting. Yes, its still needed :D

**Comment by Flyingmana** [comment]
anyway, #34  fixes the issue, maybe it helps to produce a working test

**Comment by unknownnf** [comment]
http://pastebin.com/dKKyPsKB - results, adding `"sourceReference":null` will resolve the test. The issue however only happens for unversioned vcs packages, basically tries to install the package again. 

Turned out the `add` function was called for both installs and updates, if the package has installed.json data and you try to trigger the update it throws the error.

**Comment by AydinHassan** [comment]
@bencorlett Can you confirm the packages which are falling over for you are un-versioned?.

**Comment by bencorlett** [comment]
Yep!

They’re all branch-aliased, unversioned development packages :)

Excuse the late reply, it’s been a long weekend here in Australia.

**Comment by AydinHassan** [comment]
I've figured out what the issue is here, i'll provide a PR soon

**Pull request closed without merging**